--- a/python_magnetcooling/feelpp.py
+++ b/python_magnetcooling/feelpp.py
@@ -108,7 +108,12 @@
                 )

-                Tw_inlet = TwH[i] if not isinstance(TwH[i], dict) else TwH[i]["value"]
+                if isinstance(TwH[i], dict):
+                    _csvfile = TwH[i]["filename"].replace("$cfgdir", basedir)
+                    _tw_data = pd.read_csv(_csvfile, sep=",")
+                    Tw_inlet = float(_tw_data["Tw"].iloc[0])
+                else:
+                    Tw_inlet = TwH[i]
                 T_out_guess = (
                     Tw_inlet + dTwH[i] if not isinstance(dTwH[i], dict) else None
                 )
